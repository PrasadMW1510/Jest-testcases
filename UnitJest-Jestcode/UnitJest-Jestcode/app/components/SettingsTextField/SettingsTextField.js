/**
 *
 * SettingsTextField
 *
 */

import PropTypes from 'prop-types';
import React from 'react';

import './SettingsTextField.scss';

function SettingsTextField({
  disabled,
  fieldClass: fieldClassProp,
  fieldName,
  fieldValue,
  key,
  label,
  labelClass: labelClassProp,
  labelPosition,
  mixedValue,
  onBlur,
  onChange,
}) {
  let fieldClass = `settings-text-field ${fieldClassProp}`;
  if (disabled) {
    fieldClass = fieldClass.concat(' settings-text-field--grey');
  }
  const labelClass =
    fieldValue === mixedValue
      ? labelClassProp.concat(' settings-text-field-label--grey')
      : labelClassProp;
  const keyValue = key || label;
  return (
    <span className="settings-text-field-wrapper" key={keyValue}>
      {label &&
        labelPosition === 'left' && (
          <span className={`settings-text-field-label--left ${labelClass}`}>{label}</span>
        )}
      <input
        className={fieldClass}
        disabled={disabled}
        name={fieldName}
        onBlur={onBlur}
        onChange={onChange}
        type="text"
        value={fieldValue === mixedValue ? '' : fieldValue}
      />
      {label &&
        labelPosition === 'right' && (
          <span className={`settings-text-field-label--right ${labelClass}`}>{label}</span>
        )}
    </span>
  );
}

SettingsTextField.defaultProps = {
  disabled: false,
  fieldClass: '',
  fieldName: 'settingsTextField',
  key: '',
  labelClass: '',
  labelPosition: 'left',
  mixedValue: '2',
};

SettingsTextField.propTypes = {
  disabled: PropTypes.bool.isRequired,
  fieldClass: PropTypes.string.isRequired,
  fieldName: PropTypes.string.isRequired,
  fieldValue: PropTypes.string,
  key: PropTypes.any,
  label: PropTypes.string,
  labelClass: PropTypes.string.isRequired,
  labelPosition: PropTypes.string,
  mixedValue: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
};

export default SettingsTextField;
