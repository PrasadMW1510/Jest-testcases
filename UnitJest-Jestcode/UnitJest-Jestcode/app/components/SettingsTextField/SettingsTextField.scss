.settings-text-field-label--grey {
  color: #9a9a9a;
}

.settings-text-field {
  background-color: #fff;
  margin-right: 10px;
  width: 30px;
}

.settings-text-field--grey {
  opacity: 0.6;
}
