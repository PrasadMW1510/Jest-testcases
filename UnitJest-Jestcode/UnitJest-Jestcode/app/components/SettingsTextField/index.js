/**
 *
 * SettingsFourStateTextField
 *
 */

export { default } from './SettingsTextField';
