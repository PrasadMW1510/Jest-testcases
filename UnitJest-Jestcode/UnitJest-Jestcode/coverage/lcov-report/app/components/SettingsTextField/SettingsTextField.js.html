<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/SettingsTextField/SettingsTextField.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/SettingsTextField</a> SettingsTextField.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/14</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/**<span class="cstat-no" title="statement not covered" ></span>
 *
 * SettingsTextField
 *
 */
&nbsp;
import PropTypes <span class="cstat-no" title="statement not covered" >from 'prop-types';<span class="cstat-no" title="statement not covered" ></span></span>
import React <span class="cstat-no" title="statement not covered" >from 'react';<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
<span class="cstat-no" title="statement not covered" >import './SettingsTextField.scss';<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
&nbsp;
function <span class="fstat-no" title="function not covered" >SettingsTextField(</span>{
  disabled,
  fieldClass: fieldClassProp,
  fieldName,
  fieldValue,
  key,
  label,
  labelClass: labelClassProp,
  labelPosition,
  mixedValue,
  onBlur,
  onChange,
}) {<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span></span></span></span></span></span></span></span>
  let fieldClass = <span class="cstat-no" title="statement not covered" >`settings-text-field ${fieldClassProp}`;</span>
<span class="cstat-no" title="statement not covered" >  if (disabled) {</span>
<span class="cstat-no" title="statement not covered" >    fieldClass = fieldClass.concat(' settings-text-field--grey');</span>
  }
  const labelClass =
<span class="cstat-no" title="statement not covered" >    fieldValue === mixedValue</span>
      ? labelClassProp.concat(' settings-text-field-label--grey')
      : labelClassProp;
  const keyValue <span class="cstat-no" title="statement not covered" >= key || labe</span>l;
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;span className="settings-text-field-wrapper" key={keyValue}&gt;
      {label &amp;&amp;
        labelPosition === 'left' &amp;&amp; (
          &lt;span className={`settings-text-field-label--left ${labelClass}`}&gt;{label}&lt;/span&gt;
        )}
      &lt;input
        className={fieldClass}
        disabled={disabled}
        name={fieldName}
        onBlur={onBlur}
        onChange={onChange}
        type="text"
        value={fieldValue === mixedValue ? '' : fieldValue}
      /&gt;
      {label &amp;&amp;
        labelPosition === 'right' &amp;&amp; (
          &lt;span className={`settings-text-field-label--right ${labelClass}`}&gt;{label}&lt;/span&gt;
        )}
    &lt;/span&gt;
  );
}
&nbsp;
SettingsTextField.defaultProps = {
  disabled: false,
  fie<span class="cstat-no" title="statement not covered" >ldClass: '',</span>
  fieldName: 'settingsTextField',
  key: '',
  labelClass: '',
  labelPosition: 'left',
  mixedValue: '2',
};
&nbsp;
<span class="cstat-no" title="statement not covered" >SettingsTextField.propTypes = {</span>
  disabled: PropTypes.bool.isRequired,
  fieldClass: PropTypes.string.isRequired,
  fieldName: PropTypes.string.isRequired,
  fieldValue: PropTypes.string,
  key: PropTypes.any,
  label: PropTypes.string,
  labelClass: PropTypes.string.isRequired,
  labelPosition: PropTypes.string,
  mixedValue: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,<span class="cstat-no" title="statement not covered" ></span>
};
&nbsp;
export default SettingsTextField;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 21 2018 17:06:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
