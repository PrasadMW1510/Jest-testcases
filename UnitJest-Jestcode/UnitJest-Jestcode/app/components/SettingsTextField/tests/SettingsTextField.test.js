import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import SettingsTextField from '../index';

describe('<SettingsTextField />', () => {
  let wrapper = null;
  let mockOnChange = null;

  beforeEach(() => {
    mockOnChange = jest.fn();
  });

  it('should render enabled/non-mixed correctly', () => {
    wrapper = shallow(
      <SettingsTextField
        fieldClass="fieldClass"
        fieldName="fieldName"
        fieldValue="1"
        label="label"
        labelClass="labelClass"
        labelPosition="right"
        mixedValue="-1"
        onChange={mockOnChange}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render disabled/mixed/left-label correctly', () => {
    wrapper = shallow(
      <SettingsTextField
        disabled
        fieldClass="fieldClass"
        fieldName="fieldName"
        fieldValue="-1"
        label="label"
        labelPosition="left"
        mixedValue="-1"
        onChange={mockOnChange}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
