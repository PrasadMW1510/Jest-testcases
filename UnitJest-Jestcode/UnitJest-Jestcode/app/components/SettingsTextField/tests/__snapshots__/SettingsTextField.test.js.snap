// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsTextField /> should render disabled/mixed/left-label correctly 1`] = `
<span
  className="settings-text-field-wrapper"
  key="label"
>
  <span
    className="settings-text-field-label--left  settings-text-field-label--grey"
  >
    label
  </span>
  <input
    className="settings-text-field fieldClass settings-text-field--grey"
    disabled={true}
    name="fieldName"
    onChange={[Function]}
    type="text"
    value=""
  />
</span>
`;

exports[`<SettingsTextField /> should render enabled/non-mixed correctly 1`] = `
<span
  className="settings-text-field-wrapper"
  key="label"
>
  <input
    className="settings-text-field fieldClass"
    disabled={false}
    name="fieldName"
    onChange={[Function]}
    type="text"
    value="1"
  />
  <span
    className="settings-text-field-label--right labelClass"
  >
    label
  </span>
</span>
`;
